frontend/src/app/modules/user-profile/user-profile-page/user-profile.component.ts

```diff
@@ -0,0 +1,210 @@
+import { UserProfileDto } from 'src/app/models/user/user-profile-dto';
+
+@Component({
+    selector: 'app-user-profile',
+    templateUrl: './user-profile.component.html',
+    styleUrls: ['./user-profile.component.sass'],
+})
+export class UserProfileComponent extends BaseComponent implements OnInit, OnDestroy {
+    public squirrelNotification: boolean;
+
+    public emailNotification: boolean;
+
+    public penIcon = faPen;
+
+    public currentUser: UserProfileDto;
+
+    public userNamesForm: FormGroup = new FormGroup({});
+
+    public passwordForm: FormGroup = new FormGroup({});
+
+    constructor(
+        private fb: FormBuilder,
+        private userService: UserService,
+        private notificationService: NotificationService,
+        private spinner: SpinnerService,
+    ) {
+        super();
+    }
+
+    public ngOnInit() {
+        this.fetchCurrentUser();
+    }
+
+    private fetchCurrentUser() {
+        this.spinner.show();
+
+        this.userService
+            .getUserProfile()
+            .pipe(
+                takeUntil(this.unsubscribe$),
+                finalize(() => this.spinner.hide()),
+            )
+            .subscribe(
+                (userProfile) => {
+                    this.currentUser = userProfile;
+                    this.initializeForms();
+                },
+                () => {
+                    this.notificationService.error('Failed to fetch current user');
+                },
+            );
+    }
+
+    private initializeForms() {
+        this.initUserNamesForm();
+        this.initChangePasswordForm();
+        this.initNotificationsValue();
+    }
+
+    private initUserNamesForm() {
+        this.userNamesForm = this.fb.group({
+            userName: [
+                this.currentUser.userName,
+                [Validators.required, Validators.minLength(2), Validators.maxLength(25), ValidationsFn.userNameMatch()],
+            ],
+            firstName: [
+                this.currentUser.firstName,
+                [Validators.required, Validators.minLength(2), Validators.maxLength(25), ValidationsFn.nameMatch()],
+            ],
+            lastName: [
+                this.currentUser.lastName,
+                [Validators.required, Validators.minLength(2), Validators.maxLength(25), ValidationsFn.nameMatch()],
+            ],
+        });
+    }
+
+    private initChangePasswordForm() {
+        this.passwordForm = this.fb.group({
+            currentPassword: [
+                '',
+                [
+                    Validators.required,
+                    Validators.minLength(2),
+                    Validators.maxLength(25),
+                    ValidationsFn.lowerExist(),
+                    ValidationsFn.upperExist(),
+                ],
+            ],
+            newPassword: [
+                '',
+                [
+                    Validators.required,
+                    Validators.minLength(2),
```

Comment:
Acording to specification we need at lest 6 character in the password, just so as not to rework later